autorest/azure/environments.go

```diff
@@ -78,6 +79,7 @@ var (
 		ServiceManagementVMDNSSuffix: "cloudapp.net",
 		ResourceManagerVMDNSSuffix:   "cloudapp.azure.com",
 		ContainerRegistryDNSSuffix:   "azurecr.io",
+		TokenAudience:                "https://management.azure.com/",
```

Comment:
This is a duplicate of ```ResourceManagerEndpoint```.  Why is this necessary?

Reply:
For AzureStack, the TokenAudience is different from the ResourceManagerEndpoint, Hence this is needed to support AzureStack better

Reply:
I see, it just defaults to the same value as ResourceManagerEndpoint?

Reply:
Yes, in case of AzureClouds audiences value is same as RM endpoint e.g. 
pls look at audiences value
https://management.azure.com/metadata/endpoints?api-version=1.0
https://management.usgovcloudapi.net/metadata/endpoints?api-version=1.0

in case of azure stack, the audiences value differs with each deployment. (i am not sure whether you have access to the following endpoints)
https://management.redmond.azurestack.corp.microsoft.com/metadata/endpoints?api-version=1.0
https://management.shanghai.azurestack.corp.microsoft.com/metadata/endpoints?api-version=1.0